gobblin-data-management/src/main/java/org/apache/gobblin/util/commit/SetPermissionCommitStep.java

```diff
@@ -37,18 +38,22 @@
 /**
  * An implementation of {@link CommitStep} for setting any file permissions.
  * Current implementation only sets permissions, but it is capable of setting owner and group as well.
+ * Expects a tree map of path to {@link OwnerAndPermission} ordered by highest path depth to least path depth in order to
+ * handle the edge case where setting permissions at a parent folder can lock out setting permissions on children
  */
 @Slf4j
 public class SetPermissionCommitStep implements CommitStep {
   @Getter
   Map<String, OwnerAndPermission> pathAndPermissions;
   private final URI fsUri;
   public final boolean stopOnError;
-  public static final String STOP_ON_ERROR_KEY = "stop.on.error";
+
+  public static final String SET_PERMISSION_CONFIG_PREFIX = SetPermissionCommitStep.class.getSimpleName();
+  public static final String STOP_ON_ERROR_KEY = SET_PERMISSION_CONFIG_PREFIX + "stop.on.error";
   public static final String DEFAULT_STOP_ON_ERROR = "false";
```

Comment:
apologies if I missed a code comment, but why is the default false here, but true in the other `CommitStep`?

Reply:
This step is used in the recursive copyable dataset which is set for a lot of different distcp modes such as iceberg distcp, so I didn't want to change the default behavior here. With Manifest distcp since we know that owner/group is critical given the usecase I set that default to true